src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -177,20 +176,12 @@ bool checkPositionalArguments(ASTPtr & argument, const ASTSelectQuery * select_q
             }
         }
         else
-            positional = false;
-    }
-    else if (const auto * ast_function = typeid_cast<const ASTFunction *>(argument.get()))
-    {
-        if (ast_function->arguments)
-        {
-            for (auto & arg : ast_function->arguments->children)
-                positional &= checkPositionalArguments(arg, select_query, expression);
-        }
+            return false;
     }
     else
-        positional = false;
+        return false;
 
-    return positional;
+    return true;
```

Comment:
This line is never executed

Reply:
It is executed (case when first 3 if's are true)